### PR TITLE
Remove simulation modules from opentitan synthesis flow

### PR DESCRIPTION
**Pull request by robertszczepanski**
Few modules are required only for simulation and we don't need to check if they synthesize.

### CODE REVIEW

**Comment by wsipak** [comment]
The top core that we're using currently is called `lowrisc:dv:chip_yosys_synth`.
You can see it selected here:
https://github.com/antmicro/yosys-systemverilog/blob/master/uhdm-tests/opentitan/Makefile.in#L32

The core wraps `lowrisc:systems:chip_earlgrey_yosys` and several dependencies that you're removing here.
It seem like we could get rid of the current top core (`lowrisc:dv:chip_yosys_synth`) completely (remove the file from the patch), and choose `lowrisc:systems:chip_earlgrey_yosys` as the new top (in the makefile).

File: uhdm-tests/opentitan/0001-add-Core-files-for-processing-Earlgrey-in-Yosys.patch [hunk 1]
```diff
@@ -89,17 +88,6 @@ index 000000000..0b31d6c79
...
-+      - lowrisc:dv_dpi:dmidpi
-+      - lowrisc:dv_dpi:spidpi
-+      - lowrisc:dv_dpi:usbdpi
-+      - lowrisc:dv_verilator:memutil_verilator
-+      - lowrisc:dv_verilator:simutil_verilator
-+      - lowrisc:dv:sim_sram
-+      - lowrisc:dv:sw_test_status
-+      - lowrisc:dv:dv_test_status
 +      - lowrisc:systems:chip_earlgrey_yosys
 +
```
Comment (wsipak) [inline uhdm-tests/opentitan/0001-add-Core-files-for-processing-Earlgrey-in-Yosys.patch:92]: I would suggest that we remove the file `chip_synth.core` completely from the patch, and use the core defined in the first file as the top instead.

**Review by wsipak — CHANGES_REQUESTED** [review]
The top core that we're using currently is called `lowrisc:dv:chip_yosys_synth`.
You can see it selected here:
https://github.com/antmicro/yosys-systemverilog/blob/master/uhdm-tests/opentitan/Makefile.in#L32

The core wraps `lowrisc:systems:chip_earlgrey_yosys` and several dependencies that you're removing here.
It seem like we could get rid of the current top core (`lowrisc:dv:chip_yosys_synth`) completely (remove the file from the patch), and choose `lowrisc:systems:chip_earlgrey_yosys` as the new top (in the makefile).

**Comment by robertszczepanski** [comment]
Good point, I applied suggested changes.
@wsipak PTAL

**Review by wsipak — APPROVED** [review]

**Comment by wsipak** [comment]
Looks good to me!
@robertszczepanski

**Pull request merged**